test/framework/kogitodataindex.go

```diff
@@ -15,94 +15,50 @@
 package framework
 
 import (
-	"fmt"
-	"strconv"
-
-	apps "k8s.io/api/apps/v1"
-
 	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
-	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
-	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
 	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
 	"github.com/kiegroup/kogito-cloud-operator/test/config"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// InstallKogitoDataIndexService deploy the Kogito Data Index service
-func InstallKogitoDataIndexService(namespace string, installerType InstallerType, replicas int) error {
-	GetLogger(namespace).Infof("%s install Kogito Data Index with %d replicas", installerType, replicas)
-	switch installerType {
-	case CLIInstallerType:
-		return cliInstallKogitoDataIndex(namespace, replicas)
-	case CRInstallerType:
-		return crInstallKogitoDataIndex(namespace, int32(replicas))
-	default:
-		panic(fmt.Errorf("Unknown installer type %s", installerType))
-	}
+// InstallKogitoDataIndex returns the installation service for the Kogito Data Index
+func InstallKogitoDataIndex(namespace string, installerType InstallerType, replicas int) error {
+	serviceName := getDataIndexServiceName()
+	imageTag := getDataIndexImageTag()
+	cliName := "data-index"
+	crBuilder := getDataIndexCrBuilder()
+	persistence := false
```

Comment:
can you put all of this in a structure ?
it would lower the number of arguments into `InstallService` method and would be more readable

Reply:
@sutaakar what do you think about this? I made use of structures in the first version, so this would imply to have one more type with these five fields.

Reply:
What about using KogitoDataIndex structure here? That way we can fill the values directly into the structure used to create the CR (in case of CR deployer).

Reply:
Do you mean the structure "KogitoDataIndex" in kogitodataindex_types.go? I think this type is intended to represent an object in Openshift and this method is used to actually create the object. Therefore, as the KogitoDataIndex structure is coming from Openshift (with objects, Specs and so on), I don't really like to create one instance (like a stub or dummy instance). Morever, we would still need other values like the cli name or the cr types (or builder).

Reply:
> Do you mean the structure "KogitoDataIndex" in kogitodataindex_types.go?

Yes.

> I think this type is intended to represent an object in Openshift and this method is used to actually create the object.

Yes, this type is intended to represent that resource in OpenShift. But this type is also used to create this resource. You use it too in this PR, see https://github.com/kiegroup/kogito-cloud-operator/pull/262/files#diff-080625abc1eacce2f2c206f2a6c3d4a0R53 . It serves both for the resource creation and for resource representation in OpenShift.

> Morever, we would still need other values like the cli name or the cr types (or builder).

The CLI name can be still passed as it is now, as it is the only information which cannot be retrieved from CR. cr types (Installer type?) doesn't need to be passed, it is retrieved from config. Builder doesn't have to be passed, as you pass already built resource.

Reply:
Refactored. Let me know what you think about the new changes.